scipy/io/mmio.py

```diff
@@ -281,20 +286,44 @@ def _get_symmetry(a):
         m,n = a.shape
         if m != n:
             return MMFile.SYMMETRY_GENERAL
-        issymm = 1
-        isskew = 1
+        issymm = True
+        isskew = True
         isherm = a.dtype.char in 'FD'
-        for j in range(n):
-            for i in range(j+1,n):
-                aij,aji = a[i][j],a[j][i]
-                if issymm and aij != aji:
-                    issymm = 0
-                if isskew and aij != -aji:
-                    isskew = 0
-                if isherm and aij != conj(aji):
-                    isherm = 0
-                if not (issymm or isskew or isherm):
-                    break
+        
+        if isinstance(a,spmatrix):
+            # check if number of nonzero entrys of lower and upper triangle matrix are equal
+            (row, col) = a.nonzero()
+            if (row < col).sum() != (row > col).sum():
+                return MMFile.SYMMETRY_GENERAL
+            
```

Comment:
In my opinion, the rows and columns have to be sorted with respect to the rows, too, before these can be compared with the corresponding transposed. So the code would look like this:
```python
a = a.tocoo()
rows = a.row
cols = a.col
sort_rows = np.argsort(rows)
rows = rows[sort_rows]
cols = cols[sort_rows]
sort_cols = np.argsort(cols)
rows_transpose, cols_transpose = cols[sort_cols], rows[sort_cols]
if np.any(rows != rows_transpose) or np.any(cols != cols_transpose)
    return MMFile.SYMMETRY_GENERAL
```
In addition, one would have to consider whether it makes sense to spend the n * log (n) effort for sorting. The succeeding test where each element is compared with the corresponding transposed has only a complexity of n. (n is the number of elements)